### PR TITLE
Validation model specs

### PR DESCRIPTION
**Pull request by johnie261**
Hi there,
### I did the following:

- Added  validations in the user model
-  Added  validations in the post model
- Added unit specs for all of the models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  first_user = User.create(name: 'Johnie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Tech enthusiast.',
+                           posts_counter: 0)
+
```
Comment (Deepakdanger) [inline spec/models/user_spec.rb:6]: - [ ] Kindly include some positive test case also. for ex;- 
`first_user.name` to equal `'Johnie'`

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'validations' do
+    first_user = User.create(name: 'Johnie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Tech enthusiast.',
+                             posts_counter: 1)
+    first_post = Post.create(author_id: first_user.id, title: 'Hello', text: 'This is my first post',
+                             comments_counter: 0, likes_counter: 0)
+
```
Comment (Deepakdanger) [inline spec/models/post_spec.rb:9]: - [ ] Kindly include some positive test case also. for ex:- 
- presennce of `title` should be `valid`.

File: spec/models/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'validations' do
+    first_user = User.create(name: 'Johnie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Tech enthusiast.',
+                             posts_counter: 1)
+    first_post = Post.create(author_id: first_user.id, title: 'Hello', text: 'This is my first post',
+                             comments_counter: 0, likes_counter: 0)
+    first_like = Like.create(post_id: first_post.id, author_id: first_user.id)
+
```
Comment (Deepakdanger) [inline spec/models/like_spec.rb:10]: - [ ] Kindly include some positive test case also. for ex;-

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  describe 'validations' do
+    first_user = User.create(name: 'Johnie', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Tech enthusiast.',
+                             posts_counter: 1)
+    first_post = Post.create(author_id: first_user.id, title: 'Hello', text: 'This is my first post',
+                             comments_counter: 0, likes_counter: 0)
+    first_comment = Comment.create(author_id: first_user.id, post_id: first_post.id, text: 'Hi am Johnie!')
+
```
Comment (Deepakdanger) [inline spec/models/comment_spec.rb:10]: - [ ] Kindly include some positive test case also. for ex;-

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @johnie261 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @johnie261, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡
 
✅ Descriptive PR 👍 
✅ Nice design 🔥 
✅ model validated correctly 🔥 
✅ Test passed 💯 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**